src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -450,7 +450,18 @@ class SchedulerActions(
             Done
         }.foreach { _ =>
           logger.info(s"Killing instances ${instances.map(_.instanceId)}")
-          killService.killInstances(instances, KillReason.OverCapacity)
+
+          def killInstances(instances: Seq[Instance]): Future[Done] = async {
+            val changeGoalsFuture = instances.map { i =>
+              if (i.hasReservation) instanceTracker.goalStopped(i.instanceId)
```

Comment:
Maybe not in this PR but why is this state transition logic (*if hasReservations ... else ...) here? Imho this belongs into smth. like `Instance.transition(state)` or `Instance.update(event)` method.

Reply:
I don't have an idea how to do this...  but in the future we will use Stopped also for other cases, not just residents

Reply:
If goalStopped has a side effect I'd prefer `setGoalStopped` or something